bundles/org.jkiss.utils/tests/org/jkiss/utils/BeanUtilsTest.java

```diff
@@ -0,0 +1,236 @@
+        Assert.assertEquals("setbar", BeanUtils.getSetterName("getbar"));
+        Assert.assertEquals("setbar", BeanUtils.getSetterName("hasbar"));
+    }
+
+    @Test
+    public void testGetSetMethod() {
+        Assert.assertNull(BeanUtils.getSetMethod(String.class, "size"));
+        Assert.assertNull(BeanUtils.getSetMethod(String.class, "length"));
+        Assert.assertNull(
+                BeanUtils.getSetMethod(String.class, "length", true));
+        Assert.assertNull(
+                BeanUtils.getSetMethod(String.class, "length", false));
+    }
+
+    @Test
+    public void testGetGetMethod() {
+        Assert.assertNull(BeanUtils.getGetMethod(String.class, "size"));
+        Assert.assertNull(BeanUtils.getGetMethod(String.class, "length"));
+        Assert.assertNull(
+                BeanUtils.getGetMethod(String.class, "length", true));
+        Assert.assertNull(
+                BeanUtils.getGetMethod(String.class, "length", false));
+    }
+
+    @Test
+    public void testPropertyNameToMethodName() {
+        Assert.assertEquals("Length",
+                BeanUtils.propertyNameToMethodName("length"));
+        Assert.assertEquals("Length",
+                BeanUtils.propertyNameToMethodName("Length"));
+        Assert.assertEquals("LENGTH",
+                BeanUtils.propertyNameToMethodName("lENGTH"));
+        Assert.assertEquals("LENGTH",
+                BeanUtils.propertyNameToMethodName("LENGTH"));
+    }
+
+    @Test
+    public void testMethodNameToPropertyName() {
+        Assert.assertNull(BeanUtils.methodNameToPropertyName(""));
+
+        Assert.assertEquals("g", BeanUtils.methodNameToPropertyName("G"));
+        Assert.assertEquals("get", BeanUtils.methodNameToPropertyName("Get"));
+        Assert.assertEquals("empty",
+                BeanUtils.methodNameToPropertyName("isEmpty"));
+        Assert.assertEquals("length",
+                BeanUtils.methodNameToPropertyName("getlength"));
+        Assert.assertEquals("length",
+                BeanUtils.methodNameToPropertyName("setlength"));
+    }
+
+    @Test
+    public void testIsArrayType() {
+        Assert.assertFalse(BeanUtils.isArrayType(String.class));
+    }
+
+    @Test
+    public void testIsCollectionType() {
+        Assert.assertFalse(BeanUtils.isCollectionType(String.class));
+    }
+
+    @Test
+    public void testGetCollectionType1() {
+        ParameterizedType parameterizedType =
+                PowerMockito.mock(ParameterizedType.class);
+        PowerMockito.when(parameterizedType.getActualTypeArguments())
+                .thenReturn(new Type[]{String.class});
+
+        Assert.assertEquals(String.class,
+                BeanUtils.getCollectionType(parameterizedType));
+
+        Assert.assertNull(BeanUtils.getCollectionType(null));
+    }
+
+    @Test
+    public void testGetCollectionType2() {
+        WildcardType wildcardType = PowerMockito.mock(WildcardType.class);
```

Comment:
I'm not sure `wildcardType` is the should be mocked if that's what you're testing...

Reply:
As both `wildcardType` and `ParameterizedType` are interfaces they will have to be implemented. 

So would you like me to do this instead?
```
    @Test
    public void testGetCollectionType2() {
        WildcardType wildcardType = PowerMockito.mock(WildcardType.class);
        PowerMockito.when(wildcardType.getUpperBounds())
                .thenReturn(new Type[]{String.class, Integer.class});

        ParameterizedType parameterizedType = new ParameterizedType() {
            @Override
            public Type[] getActualTypeArguments() {
                return new Type[]{wildcardType};
            }

            @Override
            public Type getRawType() {
                return null;
            }

            @Override
            public Type getOwnerType() {
                return null;
            }
        };

        Assert.assertEquals(String.class,
                BeanUtils.getCollectionType(parameterizedType));
    }
```

Reply:
If that still works, yes.

Reply:
Done